### PR TITLE
New gateway onboarding

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
References #3947
OS counterpart of https://github.com/TheThingsIndustries/lorawan-stack/pull/3304

This PR adds the new gateway onboarding which retires the old gateway registration form. The OS version of the gateway onboarding only contains manual registration (and no claiming).

image

#### Changes

- Remove old gateway registration
- Introduce new gateway onboarding flow
- Update end-to-end tests accordingly

#### Testing

Manual and end-to-end tests.

#### Notes for Reviewers
This implementation is essentially a simplified version of https://github.com/TheThingsIndustries/lorawan-stack/pull/3304
which removed all functionality related to claiming as well as detecting claimability based on the gateway EUI.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/views/gateway-add/connect.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
```
Comment (ryaplots) [inline pkg/webui/console/views/gateway-add/connect.js]: Here you are creating a connected component. Isn't it better to use react hooks?

Comment (kschiffer) [reply pkg/webui/console/views/gateway-add/connect.js]: Yes, I've refactored this to use hooks only.

File: pkg/webui/console/containers/gateway-onboarding-form/index.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        rights: ['RIGHT_GATEWAY_INFO', 'RIGHT_GATEWAY_LINK'],
+      }
+
+      return dispatch(attachPromise(createGatewayApiKey(gateway_id, key)))
+    },
+    [dispatch],
+  )
+
+  const handleRegistrationSubmit = useCallback(
+    async (values, { authenticated_identifiers, ...cleanValues }) => {
```
Comment (ryaplots) [inline pkg/webui/console/containers/gateway-onboarding-form/index.js]:
```suggestion
    async (values, cleanValues) => {
```

This part does not include claiming right? So, am I getting something wrong or there would be no `authenticated_identifiers`? I also didn't find any value with this name in the forms of this PR.

**Review by ryaplots — APPROVED** [review]
Aside from my comment, the rest looks all good to me.

File: pkg/webui/console/containers/gateway-onboarding-form/index.js (hunk 2 above)
Comment (kschiffer) [reply pkg/webui/console/containers/gateway-onboarding-form/index.js]: True. This can be removed indeed.

**Pull request merged**